### PR TITLE
Chore: Implement Feedback

### PR DESCRIPTION
**Pull request by codestaintin**
## What does this PR do?
- Feedback implementation

## Description of tasks to be completed
- [x] Implement feedback from review

## How should this be manually tested
- Clone the repo - `$ git clone https://github.com/codestaintin/gitsheet.git`
- Run ```git checkout feedback-branch```

## What are the relevant pivotal tracker stories
N/A

## Screenshot (if appropriate)
N/A

## Questions
N/A

[Finished feedback-implementation]

### CODE REVIEW

File: client/src/actions/auth/signupAction.js [hunk 1]
```diff
@@ -21,12 +25,15 @@ const signUp = (user) => dispatch => {
...
+      dispatch(signInSuccess());
       toastr.success(res.data.message);
       toastr.clear();
     })
     .catch(error => {
       if (error) {
         dispatch(signUpFailure(true));
+        toastr.error('An error occured during this operation');
+        toastr.clear();
         throw (error);
```
Comment (codrex) [inline client/src/actions/auth/signupAction.js]: Are you planning on catching this error some else? If not I don't think it is needed.

Comment (codestaintin) [reply client/src/actions/auth/signupAction.js]: Thanks for the feedback, this has been updated! @codrex

File: client/src/components/Home.jsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import PropTypes from 'prop-types';
 import Loader from 'react-loader-spinner';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import Header from './common/Header';
+// import Header from './common/Header';
```
Comment (codrex) [inline client/src/components/Home.jsx]: Remove dead code

File: client/src/components/Home.jsx [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class Home extends React.Component {
     const { categories, isLoading } = this.props;
     return (
       <Fragment>
-        <Header sidebarOpen={this.state.sidebarOpen} />
+        {/* <Header sidebarOpen={this.state.sidebarOpen} /> */}
```
Comment (codrex) [inline client/src/components/Home.jsx]: Remove dead code

File: client/src/components/common/AuthHeader.jsx [hunk 4]
```diff
@@ -2,11 +2,11 @@ import React, { Fragment } from 'react';
 import { Link } from 'react-router-dom';
 import PropTypes from 'prop-types';
 
-const AuthHeader = ({ logout }) => (
+const AuthHeader = ({ logout, sidebarOpen }) => (
```
Comment (codrex) [inline client/src/components/common/AuthHeader.jsx:5]: Having two headers is not the best approach. Please see a refactored implementation for better insight
https://repl.it/@codrexy/DisguisedWetTwintext?language=reactjs

File: client/src/actions/auth/signupAction.js [hunk 5]
```diff
@@ -13,21 +13,28 @@ const signUpFailure = (bool) => ({
...
     .then((res) => {
       const { token } = res.data;
-      if (decodeToken(token)) {
+      const validToken = Object.keys(decodeToken(token)).includes('_id');
+      if (validToken) {
         window.localStorage.setItem('token', token);
       }
-      dispatch(signUpSuccess(true));
+      dispatch(signUpSuccess(validToken));
+      dispatch(signInSuccess(validToken));
```
Comment (codrex) [inline client/src/actions/auth/signupAction.js]: I thought this thunk is for sign in, why do we have `dispatch(signUpSuccess(validToken));`?

File: client/src/actions/actionTypes.js [hunk 6]
```diff
@@ -8,5 +8,6 @@ export default keymirror({
   LOG_OUT_SUCCESS: null,
   LOAD_CHEATS_SUCCESS: null,
   LOAD_CHEATS_FAILURE: null,
-  IS_LOADING: null
+  IS_LOADING: null,
+  IS_AUTHENTICATED: null
```
Comment (codrex) [inline client/src/actions/actionTypes.js]: `SIGNUP_SUCCESSFUL: null,
   SIGNUP_UNSUCCESSFUL: null,
   SIGNIN_SUCCESSFUL: null,
   SIGNIN_UNSUCCESSFUL: null,
`
The above is not necessary. `IS_AUTHENTICATED` should take care of both sign in and sign up

**Review by codrex — CHANGES_REQUESTED** [review]

**Comment by codrex** [comment]
@codestaintin please also attend to the following;
screenshot 2018-12-16 at 3 05 26 pm
Side bar is bellow the navbar; adding a position of fixed to the sidebar should fix it.

screenshot 2018-12-16 at 3 06 30 pm
Having `copied` remain on the screen after a copy operation is not good UX. A better approach would be to have the tooltip leave the screen after a given period.

screenshot 2018-12-16 at 3 38 18 pm

Having your store structured this way is not the best approach. signupReducer and signinReducer should be one reducer. Combining both reducers as authReducer will DRY up your implementation.

File: client/src/actions/auth/signupAction.js (hunk 5 above)
Comment (codestaintin) [reply client/src/actions/auth/signupAction.js]: The validToken is a truthy value, it returns true or false based on a computed value of the validity of the token

Comment (codrex) [reply client/src/actions/auth/signupAction.js]: yeah, but this thunk function is for `signin` and not `signup`. So I am wondering why we have `signUpSuccess dispatch` in a `signin`  thunk

Comment (codestaintin) [reply client/src/actions/auth/signupAction.js]: Oh, I get what you mean

**Review by codrex — APPROVED** [review]

**Pull request merged**